index.bs

```diff
@@ -234,6 +238,110 @@ The <dfn attribute for=FileSystemHandle>kind</dfn> getter steps are to return
+      "{{NotFoundError}}" {{DOMException}} and abort.
+  1. Let |destination_entry| be an [=/entry=] in |new_directory|'s [=directory entry/children=].
+    1. Set |destination_entry|'s [=entry/name=] to |new_name|.
+    1. If |entry| is [=the same as=] |destination_entry|,
+       [=/resolve=] |result| with `undefined`.
+     1. If |destination_entry| exists in the underlying file system:
+       1. [=/Reject=] |result| with an "{{InvalidModificationError}}" {{DOMException}}
+           and abort.
+
+  1. Let |accessSource| be the result of running |entry|'s
+     [=entry/request access=] given "`readwrite`".
+  1. If |accessSource| is not "{{PermissionState/granted}}", reject |result| with a
+     "{{NotAllowedError}}" {{DOMException}} and abort.
+  1. If |new_directory| is not [=entry/parent=]:
+    1. Let |accessDest| be the result of running |new_directory|'s
+       [=entry/request access=] given "`readwrite`".
+    1. If |accessDest| is not "{{PermissionState/granted}}", reject |result| with a
+       "{{NotAllowedError}}" {{DOMException}} and abort.
+
+  1. Let |lockResultSource| be the result of [=file entry/lock/take|taking a lock=]
+     with "`exclusive`" on |entry|.
+  1. If |lockResultSource| is false, [=reject=] |result| with a
+     "{{NoModificationAllowedError}}" {{DOMException}} and abort.
+  1. Let |lockResultDestination| be the result of [=file entry/lock/take|taking a lock=]
+     with "`exclusive`" on |destination_entry|.
+  1. If |lockResultDestination| is false, [=reject=] |result| with a
+     "{{NoModificationAllowedError}}" {{DOMException}} and abort.
+
+  1. If |new_directory| does not exist in the underlying file system:
+    1. [=/Reject=] |result| with a "{{NotFoundError}}" {{DOMException}} and abort.
+  1. If |entry| does not exist in the underlying file system:
+      1. [=/Reject=] |result| with a "{{NotFoundError}}" {{DOMException}} and abort.
+  1. If |entry| is a [=directory entry=]:
+    1. If |entry| is in the Origin Private File System and |destination_entry| is not:
+      1. [=set/For each=] of |entry|'s [=directory entry/children=]:
+        1. Run [=implementation-defined=] malware scans and safe browsing checks.
+           If these checks fail, [=/reject=] |result| with an
+           "{{AbortError}}" {{DOMException}} and abort.
+  1. If |entry| is a [=file entry=]:
+    1. If |destination_entry| is not in the Origin Private File System:
+      1. Run [=implementation-defined=] malware scans and safe browsing checks.
+         If these checks fail, [=/reject=] |result| with an
+         "{{AbortError}}" {{DOMException}} and abort.
+
+  1. Attempt to move |entry| to |destination_entry| in the underlying file system.
+    1. If the underlying move operation throws an exception, [=/reject=]
+       |result| with that exception and abort.
+
+     Issue(68): Better specify what possible exceptions this could throw.
+
+  1. Set |entry|'s [=entry/name=] to |new_name|.
```

Comment:
note: we should also specify that |entry|'s parent is now |new_directory|. Not sure how this should be worded until we have resolution on #59